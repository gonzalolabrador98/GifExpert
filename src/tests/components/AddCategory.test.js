import { shallow } from "enzyme"
import React from "react"
import { AddCategory } from "../../components/AddCategory"


describe(' Pruebas AddCategory.js', () => {
    
 const setCategories = jest.fn();
 let wrapper=shallow(<AddCategory setCategories={setCategories}/>)

 beforeEach ( ()=> {

    jest.clearAllMocks();
    wrapper=shallow(<AddCategory setCategories={setCategories}/>)

 })

    test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    })
    
    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input')
        const value = 'Hola Mundo';

        input.simulate('change', {target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value)
        })


    test('NO debe postear la informacion con submit', () => {
            
            wrapper.find('form').simulate('submit', {preventDefault(){}})

            expect(setCategories).not.toHaveBeenCalled();
        })
        
    test('debe postear la informacion con submit', () => {
        
        const value = 'Hola';
        
        wrapper.find('input').simulate('change', {target:{value}})

        wrapper.find('form').simulate('submit', { preventDefault(){}});
              
        expect(setCategories).toHaveBeenCalled();

        expect(wrapper.find('input').prop('value')).toBe('');
    })
    

})
