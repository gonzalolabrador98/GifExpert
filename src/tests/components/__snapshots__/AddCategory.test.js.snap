// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas AddCategory.js debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
